src/wallet/crypter.cpp

```diff
@@ -551,7 +551,7 @@ bool CCryptoKeyStore::GetSaplingSpendingKey(const libzcash::SaplingFullViewingKe
 bool CCryptoKeyStore::EncryptKeys(CKeyingMaterial& vMasterKeyIn)
 {
     {
-        LOCK2(cs_KeyStore, cs_SpendingKeyStore);
+        LOCK(cs_KeyStore);
         if (!mapCryptedKeys.empty() || IsCrypted())
             return false;
 
```

Comment:
`fUseCrypto` is written inside the lock (expand just below this comment), but read outside the lock, for example at https://github.com/zcash/zcash/blob/0901007ed5743db5c818f8e1d0a7406b6207bf7c/src/wallet/crypter.h#L158-L173 and in other calls to `IsCrypted()`, for example from `GetKeys` and `GetSproutPaymentAddresses` (defined in the header).

If a lock is needed for concurrent writes then all reads must also be locked; otherwise the behaviour is undefined. It's actually unclear that there's any benefit to having an accessor method, since the temptation would be to put the locking there but that is too fine-grained.

Reply:
I see that `IsCrypted()` is used from outside the class, which means that it must explicitly lock. In that case, accesses to `fUseCrypto` from inside the class should probably be bare accesses with the lock already held.